RHDEVDOCS-5845: Post GA changes in the GitOps 1.14 content (#82209)
openshift/openshift-docs: modules/gitops-release-notes-1-14-0.adoc

* With this update, you can implement an alternative sharding algorithm that utilizes consistent hashing with bounded loads. The algorithm ensures a uniform distribution of managed clusters across shards while minimizing the need for reshuffling during configuration changes, such as adjustments in the number of managed clusters or shards. link:https://issues.redhat.com/browse/GITOPS-3584[GITOPS-3584]

> * With this update, {gitops-title} provides support for the `sidecar` and `init` containers for the Server and Application Controller components. You can configure the `sidecar` and `init` containers by using the `spec.server.sidecarContainers`, `spec.controller.sidecarContainers`, `spec.server.initContainers` and `spec.controller.initContainers` fields in the Argo CD CR. link:https://issues.redhat.com/browse/GITOPS-5010[GITOPS-5010]

Author: Fixed.

+
.Example 
[source,YAML]
----
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  server:
    initContainers:
    - name: <name of the initContainer> #<1>
      image: <image name of the initContainer> #<2>
    sidecarContainers:
    - name: <name of the sidecarContainer> #<3>

>       image: <image name of the sidecarContainer> #<4>

Author: Fixed.

* With this update, you can implement an alternative sharding algorithm that utilizes consistent hashing with bounded loads. The algorithm ensures a uniform distribution of managed clusters across shards while minimizing the need for reshuffling during configuration changes, such as adjustments in the number of managed clusters or shards. link:https://issues.redhat.com/browse/GITOPS-3584[GITOPS-3584]

* With this update, {gitops-title} provides support for the `sidecar` and `init` containers for the Server and Application Controller components. You can configure the `sidecar` and `init` containers by using the `spec.server.sidecarContainers`, `spec.controller.sidecarContainers`, `spec.server.initContainers` and `spec.controller.initContainers` fields in the Argo CD CR. link:https://issues.redhat.com/browse/GITOPS-5010[GITOPS-5010]
+
.Example 
[source,YAML]
----
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  server:
    initContainers:
    - name: <name of the initContainer> #<1>
      image: <image name of the initContainer> #<2>

>     server:

Reviewer: isnt this line wrong?

Reviewer 2: hey @Dhruv-Soni11, yeah that's a typo. You don't need `server` here.

Pasting the correct example for your reference.
```
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  server:
    initContainers:
    - name: <name of the initContainer> #(1)
      image: <image name of the initContainer> #(2)
    sidecarContainers:
    - name: <name of the sidecarContainer> #(3)
      image: <image name of the sidecarContainer> #(4)
```

Author: Fixed.

* With this update, you can implement an alternative sharding algorithm that utilizes consistent hashing with bounded loads. The algorithm ensures a uniform distribution of managed clusters across shards while minimizing the need for reshuffling during configuration changes, such as adjustments in the number of managed clusters or shards. link:https://issues.redhat.com/browse/GITOPS-3584[GITOPS-3584]

> * With this update, {gitops-title} provides support for the `sidecar` and `initcar` containers for the Server and Application Controller components. You can configure these fields by using the `spec.server.sidecarContainers` and `spec.controller.sidecarContainers` for the `sidecar` container and the `spec.server.initContainers` and `spec.controller.initContainers` for the `initcar` container. link:https://issues.redhat.com/browse/GITOPS-5010[GITOPS-5010]

Author: Fixed.

* Before this update, the {gitops-title} Operator included a race condition where the OpenShift Service CA created the TLS secret before the default TLS termination policy was set. As a result, the default policy was set as `Passthrough` instead of `Reencrypt`. This update fixes this race condition by verifying whether the secret was created by the OpenShift Service CA, ensuring that `Reencrypt` is used as the correct default policy. link:https://issues.redhat.com/browse/GITOPS-4947[GITOPS-4947]

> * Before this update, you could not login to the Argo CD web console UI after a ROSA cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4801[GITOPS-4358]

Reviewer 3: ```suggestion
* Before this update, you could not login to the Argo CD web console UI after a ROSA cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]
```

Author: Fixed.

* Before this update, the {gitops-title} Operator included a race condition where the OpenShift Service CA created the TLS secret before the default TLS termination policy was set. As a result, the default policy was set as `Passthrough` instead of `Reencrypt`. This update fixes this race condition by verifying whether the secret was created by the OpenShift Service CA, ensuring that `Reencrypt` is used as the correct default policy. link:https://issues.redhat.com/browse/GITOPS-4947[GITOPS-4947]

> * Before this update, you could not log in to the Argo CD web console UI after a ROSA cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]

Reviewer 4: ```suggestion
* Before this update, you could not log in to the Argo CD web console UI after a {product-rosa} cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]
```

Author: Fixed.

* Before this update, the {gitops-title} Operator included a race condition where the OpenShift Service CA created the TLS secret before the default TLS termination policy was set. As a result, the default policy was set as `Passthrough` instead of `Reencrypt`. This update fixes this race condition by verifying whether the secret was created by the OpenShift Service CA, ensuring that `Reencrypt` is used as the correct default policy. link:https://issues.redhat.com/browse/GITOPS-4947[GITOPS-4947]

> * Before this update, you could not login to the Argo CD web console UI after a ROSA cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]

Reviewer 3: ```suggestion
* Before this update, you could not log in to the Argo CD web console UI after a ROSA cluster, that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]
```

Author: Fixed.

<1> Specifies the name of the ApplicationSet GitLab SCM Provider.
<2> Specifies the namespace where the ApplicationSet GitLab SCM Provider is defined.
+
For more information, see link:https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#add-self-signed-tls-certificate-for-gitlab-scm-provider-to-applicationsets-controller[Add Self-signed TLS Certificate for Gitlab SCM Provider to ApplicationSet Controller].

[id="breaking-change-1-14-0_{context}"]
== Breaking change

[id="cluster-secrets-with-empty-values_{context}"]
=== Cluster secrets with empty secret values

> * WHen you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that correspondeds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, you can verify whether the project field of an application aligns with the project field of the cluster secret. As a result, if a cluster secret is designated for project-a, an application assigned to project-b will be unable to access that secret. To use applications across multiple projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]

Reviewer 5: Typo - `WHen` should be When

Reviewer 5: `no longer be classified as a global secret`

Author: Fixed.

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that matches the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets scoped to an application's project will be considered for matching a repository URL. As a result, if a cluster secret is designated to project A, an application assigned to project B will be unable to access that secret. To allow applications across various projects to use cluster secrets, you must remove the `project` field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]
> 
> For more information, see link:https://argo-cd.readthedocs.io/en/latest/operator-manual/upgrading/2.11-2.12/#cluster-secret-scoping-changes[Cluster secret scoping changes].

Reviewer 4: - Suggest breaking this up for readability. Use the `+` between lines to make sure everything is tacked on (missing in the "For more information" line as drafted).
- Should avoid using `Applications` and `ApplicationSets` as nouns. I am not familiar with this product area, but would something like "`Application` and `ApplicationSet` resources" work?
- Also recommend moving the bug link to the end to match usual format.

```suggestion
* When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value are no longer classified as global secrets applicable to all `Applications` and `ApplicationSets`. 
+
In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that matches the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets scoped to an application's project will be considered for matching a repository URL. 
+
As a result, if a cluster secret is designated to project A, an application assigned to project B will be unable to access that secret. To allow applications across various projects to use cluster secrets, you must remove the `project` field. 
+
For more information, see link:https://argo-cd.readthedocs.io/en/latest/operator-manual/upgrading/2.11-2.12/#cluster-secret-scoping-changes[Cluster secret scoping changes]. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]
```

Author: Fixed.

<1> Specifies the name of the ApplicationSet GitLab SCM Provider.
<2> Specifies the namespace where the ApplicationSet GitLab SCM Provider is defined.
+
For more information, see link:https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#add-self-signed-tls-certificate-for-gitlab-scm-provider-to-applicationsets-controller[Add Self-signed TLS Certificate for Gitlab SCM Provider to ApplicationSet Controller].

[id="breaking-change-1-14-0_{context}"]
== Breaking change

[id="cluster-secrets-with-empty-values_{context}"]
=== Cluster secrets with empty secret values

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value are no longer classified as global secrets applicable to all `Applications` and `ApplicationSets`. 

Reviewer 3: ```suggestion
* When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value are no longer classified as global secrets applicable to all `Applications` and `ApplicationSets` resources. 
```
Check with SME once for technical accuracy for all my suggestions. If everything checks okay, go ahead and please implement them.

+

> In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that matches the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets scoped to an application's project will be considered for matching a repository URL. 

Reviewer 3: ```suggestion
In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` resource could use a cluster secret that matches the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets scoped to an application's project will be considered for matching a repository URL. 
```

Author: I've checked with the SME and he is in agreement with the modification suggestion. Fixed.

Author: Fixed.

<1> Specifies the name of the ApplicationSet GitLab SCM Provider.
<2> Specifies the namespace where the ApplicationSet GitLab SCM Provider is defined.
+
For more information, see link:https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#add-self-signed-tls-certificate-for-gitlab-scm-provider-to-applicationsets-controller[Add Self-signed TLS Certificate for Gitlab SCM Provider to ApplicationSet Controller].

[id="breaking-change-1-14-0_{context}"]
== Breaking change

[id="cluster-secrets-with-empty-values_{context}"]
=== Cluster secrets with empty secret values

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secret applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that corresponds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, cluster secrets that are scoped to an application's project can be considered for matching a `repo URL`. As a result, if a cluster secret is designated for project-a, an application assigned to project-b will be unable to access that secret. To use applications across multiple projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]

Reviewer 5: if put in quotes `repo URL` is not correct. it should either be `repoUrl` we can also mention as `repository URL ` without any quotes.

Author: Fixed it as 'repoURL'.

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secret applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that correspondeds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, you can verify whether the project field of an application aligns with the project field of the cluster secret. As a result, if a cluster secret is designated for project-a, an application assigned to project-b will be unable to access that secret. To use applications across multiple projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]

Reviewer 5: typo: `correspondeds`

Reviewer 5: Suggesting to reword as follows: Starting from {gitops-title} v1.14, cluster secrets that are scoped to the application's project alone would be considered for finding a matching  repo URL.

Author: Fixed.

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that corresponds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets that are scoped to an application's project will be considered for matching a repository URL. As a result, if a cluster secret is designated for example to project A, an application assigned to project B will be unable to access that secret. To enable the use of applications across various projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]

Reviewer 3: ```suggestion
* When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that matches the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets scoped to an application's project will be considered for matching a repository URL. As a result, if a cluster secret is designated to project A, an application assigned to project B will be unable to access that secret. To allow applications across various projects to use cluster secrets, you must deselect the `project` field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]
```

Author: Fixed.

> * When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that corresponds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets that are scoped to an application's project will be considered for matching a repository URL. As a result, if a cluster secret is designated for example to project A, an application assigned to project B will be unable to access that secret. To enable the use of applications across various projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]

Reviewer 3: ```suggestion
* When you upgrade to {gitops-title} v1.14, cluster secrets with an empty project value will no longer be classified as global secrets applicable to all `Applications` and `ApplicationSets`. In previous versions of {gitops-shortname}, any `Application` or `ApplicationSet` could use a cluster secret that corresponds to the URL specified in the `repoUrl` field. Starting from {gitops-title} v1.14, only cluster secrets that are scoped to an application's project will be considered for matching a repository URL. As a result, if a cluster secret is designated for example to project A, an application assigned to project B will be unable to access that secret. To enable the use of applications across various projects for cluster secrets, you must deselect the project field. link:https://issues.redhat.com/browse/GITOPS-5623[GITOPS-5623]
```

@anandf and @Dhruv-Soni11 what do we mean when we say unset the  `project` field here? You have used deselect here. Where is the action happening? YAML file? web console UI or the argo CD UI?

Author: Hi @Srivaralakshmi - I have had a detailed discussion with @anandf and due to the unavailability of a cluster, he explained the scenario on call. Basically, in this scenario, we have to remove this field from the YAML as per his information.
